### DISCUSSION TITLE
Bind to a single Realm object

### DISCUSSION THREAD

**Discussion by Jean-RobertLaffineur** [realm/realm-dotnet#3500] [Q&A]
Hi,

I need to bind a single Realm object to my UI (MAUI dotnet 8) to display its properties in my application. The [documentation](https://www.mongodb.com/docs/realm/sdk/dotnet/model-data/data-binding/#binding-to-a-single-realm-object) gives a clue at how this can be done:

```
public Employee Employee123 { get; }
...
Employee123 = realm.All<Employee>()
    .FirstOrDefault(e => e.EmployeeId == "123");
```

This works fine for an object that exists for the whole UI lifecycle, but in my case, the object I need to bind to is the result of a query that sometimes gives one object and sometimes none. My UI should display the object when there is one, and some empty fields otherwise.

So far I have implemented it using an `IQueryable` property containing the result of the query:

```
public IQueryable<MyObject> MyObjects { get; set; }

MyObjects = Realm.All<MyObject>().Where(...);
```
... and a binding value converter that converts the `IQueryable`to a single object:

```
public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
{
    if (value != null && value is IQueryable<MyObject> myObjects)
    {
        // HACK: using myObjects.Any() or myObjects.Count() crashes with a System.NotSupportedException
        if (((Realms.RealmCollectionBase<MyObject>)myObjects).Count != 0 && myObjects.First().IsValid)
            return myObjects.FirstOrDefault();
    }

    return null;
}
```

I feel like this is not the right way to do it, as I sometimes get invalid objects issues and object's PropertyChanged that suddenly stop being fired, but the documentation is very limited in terms of examples. Any idea?

**Comment by peppy** [comment]
You should be able to use the `SubscribeForNotifications` flow for this:

```csharp
updateRealm.All<MyObject>().Where(...).SubscribeForNotifications(callback);

void callback(IRealmCollection<BeatmapSetInfo> sender, ChangeSet? changes)
{
        if (sender.Any())
        {
            // display item
        }
        else
        {
            // display placeholder
        }
}
```

The callback will fire once after the query completes (asynchronously) and then once more when the query changes. In your example you say it's one-or-none so you don't even need to check the `ChangeSet`.

**Reply by Jean-RobertLaffineur** [reply]
Thank you. I had not thought about making the link between the property and the query this way.

I will experiment with this and let you know.
